SPEC.md

```diff
@@ -8,80 +8,243 @@ Rough consensus on this spec has been reached, and implementations exist in Phan
+  - The wallet must ignore the [`feePayer`](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html#feePayer) in the transaction and set the `feePayer` to the `account` in the request.
+  - The wallet must ignore the [`recentBlockhash`](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html#recentBlockhash) in the transaction and set the `recentBlockhash` to the [latest blockhash](https://solana-labs.github.io/solana-web3.js/classes/Connection.html#getLatestBlockhash).
+
+If the transaction [`signatures`](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html#signatures) are nonempty:
+  - The application must set the [`feePayer`](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html#feePayer) to the [public key of the first signature](https://solana-labs.github.io/solana-web3.js/modules.html#SignaturePubkeyPair).
+  - The application must set the [`recentBlockhash`](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html#recentBlockhash) to the [latest blockhash](https://solana-labs.github.io/solana-web3.js/classes/Connection.html#getLatestBlockhash).
+  - The wallet must not set the  [`feePayer`](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html#feePayer) and [`recentBlockhash`](https://solana-labs.github.io/solana-web3.js/classes/Transaction.html#recentBlockhash).
+  - The wallet must verify the signatures, and if any are invalid, the wallet must reject the transaction as **malformed**.
+
+The wallet must only sign the transaction with the `account` in the request, and must do so only if a signature for the `account` in the request is expected.
+
+If any signature except a signature for the `account` in the request is expected, the wallet must reject the transaction as **malicious**.
+
+The application may also include an optional `message` field in the response body:
+```json
+{"message":"<message>","transaction":"<transaction>"}
 ```
-solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN?amount=0.01&spl-token=EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v&label=Michael&message=Thanks%20for%20all%20the%20fish&memo=OrderId5678
+
+The `<message>` value must be a UTF-8 string that describes the nature of the transaction response.
+
+For example, this might be the name of an item being purchased, a discount applied to the purchase, or a thank you note. The wallet should display the value to the user.
+
+The wallet and application should allow additional fields in the request body and response body, which may be added by future specification.
+
+### Example
+
+##### URL describing a transaction request.
+```
+solana:https://mvines.com/solana-pay
```

Comment:
I'd recommend `example.com` for the URLs, rather than a parked domain that's currently available for purchase.

Reply:
Yeah, that's probably a good idea.

Reply:
Fixed!